db/src/db.rs

```diff
@@ -758,41 +758,66 @@ impl DB {
         Ok(())
     }
 
-    /// Update the current partition map materialized view.
-    // TODO: only update changed partitions.
-    pub fn update_partition_map(&self, conn: &rusqlite::Connection) -> Result<()> {
-        let values_per_statement = 2;
-        let max_partitions = ::SQLITE_MAX_VARIABLE_NUMBER / values_per_statement;
-        if self.partition_map.len() > max_partitions {
-            bail!(ErrorKind::NotYetImplemented(format!("No more than {} partitions are supported", max_partitions)));
-        }
+    /// Transact the given `entities` against the given SQLite `conn`, using the metadata in
+    /// `self.DB`.
+    ///
+    /// This approach is explained in https://github.com/mozilla/mentat/wiki/Transacting.
+    // TODO: move this to the transactor layer.
+    pub fn transact<I>(&mut self, conn: &rusqlite::Connection, entities: I) -> Result<TxReport> where I: IntoIterator<Item=Entity> {
+        // Eventually, this function will be responsible for managing a SQLite transaction.  For
+        // now, it's just about the tx details.
+
+        let tx_instant = now(); // Label the transaction with the timestamp when we first see it: leading edge.
+        let tx_id = self.partition_map.allocate_entid(":db.part/tx");
 
-        // Like "UPDATE parts SET idx = CASE WHEN part = ? THEN ? WHEN part = ? THEN ? ELSE idx END".
-        let s = format!("UPDATE parts SET idx = CASE {} ELSE idx END",
-                        repeat("WHEN part = ? THEN ?").take(self.partition_map.len()).join(" "));
+        self.create_temp_tables(conn)?;
 
-        let params: Vec<&ToSql> = self.partition_map.iter().flat_map(|(name, partition)| {
-            once(name as &ToSql)
-                .chain(once(&partition.index as &ToSql))
-        }).collect();
+        let mut tx = Tx::new(self, conn, tx_id, tx_instant);
+        tx.transact_entities(entities)
+    }
+}
 
-        // TODO: only cache the latest of these statements.  Changing the set of partitions isn't
-        // supported in the Clojure implementation at all, and might not be supported in Mentat soon,
-        // so this is very low priority.
-        let mut stmt = conn.prepare_cached(s.as_str())?;
-        stmt.execute(&params[..])
-            .map(|_c| ())
-            .chain_err(|| "Could not update partition map")
+/// Update the current partition map materialized view.
```

Comment:
The diff has totally broken down at this point.
It looks like there's some moving around of functions -> Just transact() into PartitionMap?